cmds/validate/validate.go

```diff
@@ -43,18 +43,50 @@ var (
 		"SHA3_512":  crypto.SHA3_512,
 	}
 
-	armored = flag.Bool("a", false, "signature is ASCII armored")
-	sumfile = flag.String("i", "", "checksum file")
-	alg     = flag.String("alg", "MD5", "md5sum")
-	verbose = flag.Bool("v", false, "verbose")
-	debug   = func(string, ...interface{}) {}
+	armored    = flag.Bool("a", false, "signature is ASCII armored")
+	sumfile    = flag.String("i", "", "checksum file")
+	alg        = flag.String("alg", "", "algorithms to check")
+	verbose    = flag.Bool("v", false, "verbose")
+	debug      = func(string, ...interface{}) {}
+	try, tried []string
 )
 
+func init() {
+	for v := range algs {
+		try = append(try, v)
+	}
+}
+
+func one(n string, b []byte, sig string) bool {
+	defer func() {
+		if r := recover(); r != nil {
+			log.Printf("Hash %v did not get linked in: ", n, r)
```

Comment:
You've given two args to `Printf` but used only one, go will complain about that, isn't it?